### PR TITLE
[SDK]Enable telemetry for CLI & default not logging

### PR DESCRIPTION
**Pull request by D-W-**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

This PR adds telemetry for public API calls and default not logging.
Currently only public API telemetries are recorded.
Used test app insight to store telemetry
If user configured share telemetry in extension, we'll respect it in CLI.

Sample columns
image

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_cli/_configuration.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+import json
+import os.path
+import uuid
+from pathlib import Path
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_cli/_configuration.py:7]: I think you already check in this, can we refresh the PR to only show the new changes?

File: src/promptflow/promptflow/_telemetry/telemetry.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        :rtype: Dict
+        """
+        return {}
+
+
+def is_telemetry_enabled():
+    """Check if telemetry is enabled. User can enable telemetry by
+    1. setting environment variable TELEMETRY_ENABLED to true.
+    2. running `pf config set cli.telemetry_enable=true` command.
+    If None of the above is set, will prompt an input to ask user to enable telemetry.
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_telemetry/telemetry.py:32]: where do we prompt for this?

File: .cspell.json [hunk 3]
```diff
@@ -126,7 +126,8 @@
     "aggr",
     "tcsetattr",
     "pysqlite",
-    "AADSTS700082"
+    "AADSTS700082",
+    "levelno"
```
Comment (wangchao1230) [inline .cspell.json:130]: Note we should add telemetry notic:
○ Telemetry Notice. If the release has code that enables Microsoft to collect telemetry, also include in the repository:  [Task 2573599](https://msdata.visualstudio.com/Vienna/_workitems/edit/2573599): Telemetry Notice
			§ instructions to turn off the telemetry;
this [telemetry/data collection notice](https://docs.opensource.microsoft.com/releasing/general-guidance/telemetry) in a the README.md file.

Comment (wangchao1230) [reply .cspell.json:130]: one example: https://github.com/microsoft/onefuzz/blob/58da7b431e0b307155b508eeee5922f23b209c46/README.md?plain=1#L79

File: src/promptflow/promptflow/_telemetry/telemetry.py (hunk 2 above)
Comment (D-W-) [reply src/promptflow/promptflow/_telemetry/telemetry.py:32]: Good catch, we won't do this accoding to last sync, let me change the doc string.

File: src/promptflow/promptflow/_cli/_configuration.py (hunk 1 above)
Comment (D-W-) [reply src/promptflow/promptflow/_cli/_configuration.py]: Removed

File: .cspell.json (hunk 3 above)
Comment (D-W-) [reply .cspell.json:129]: Updated and referenced `azure-cli`'s behavior; 
image

File: src/promptflow/promptflow/_telemetry/logging_handler.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+# ---------------------------------------------------------
+import logging
+import platform
+
+from opencensus.ext.azure.log_exporter import AzureEventHandler
+
+from promptflow._cli._user_agent import USER_AGENT
+from promptflow._sdk._configuration import Configuration
+
+# b4ff2b60-2f72-4a5f-b7a6-571318b50ab2
```
Comment (zhengfeiwang) [inline src/promptflow/promptflow/_telemetry/logging_handler.py:12]: duplicate comment?

**Review by zhengfeiwang — DISMISSED** [review]

File: src/promptflow/promptflow/_telemetry/logging_handler.py (hunk 4 above)
Comment (D-W-) [reply src/promptflow/promptflow/_telemetry/logging_handler.py:12]: Good catch

**Review by zhengfeiwang — DISMISSED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Review by zhengfeiwang — DISMISSED** [review]

**Review by brynn-code — APPROVED** [review]

**Review by crazygao — APPROVED** [review]
LGTM

**Pull request merged**